dbms/src/TiDB/Schema/SchemaBuilder.cpp

```diff
@@ -86,16 +91,28 @@ void SchemaBuilder<Getter, NameMapper>::applyCreateTable(DatabaseID database_id,
     }
 
     // If table is partition table, we will create the logical table here.
-    // Because we get the table_info, so we can ensure new_db_info will not be nullptr.
     auto new_db_info = getter.getDatabase(database_id);
-    applyCreateStorageInstance(new_db_info, table_info);
+    if (new_db_info == nullptr)
+    {
+        // the database has been dropped
+        LOG_INFO(
+            log,
+            "database is not exist in TiKV, may have been dropped, applyCreateTable is ignored, database_id={} "
+            "table_id={}",
+            database_id,
+            table_id);
+        return;
+    }
+    applyCreateStorageInstance(new_db_info, table_info, get_by_mvcc);
```

Comment:
add some comments to explain why creating logical table need to create storage instance.

Reply:
added